pydantic/mypy.py

```diff
@@ -1,6 +1,7 @@
 from configparser import ConfigParser
 from typing import Any, Callable, Dict, List, Optional, Set, Tuple, Type as TypingType, Union
 
+import toml
```

Comment:
This PR does not add `toml` to `pydantic`'s dependencies, so this import and its use later in the file should be more cautious as written, i.e.

```suggestion
try:
    import toml
except ImportError:
    import warnings
    warnings.warn("'toml'  module not installed, will not be able to read configuration from pyproject.toml without it")
    toml = None  # and then check `if toml and options.config_file.endswith('.toml'):` during usage
```

However, note that [`tomli` is part of `mypy`'s dependencies](https://github.com/python/mypy/blob/5e73cd7b0dc9b871716acf0086b2ca0376fcff39/setup.py#L196) and should therefore already be present on the user's system. I'd suggest rewriting to use it.

Reply:
That seems reasonable to me. For practical purposes the plugin file would only be read in the case of mypy's use, which as you noted is a hard dependency for `mypy` itself and thus would be present, but it doesn't hurt anything to check here.

Reply:
resolved in 3aaa75a5ecc60691a99ca0e19e9978980ccb5967.

Reply:
Point of clarification: `mypy` depends on [`tomli`](https://pypi.org/project/tomli/), which is a different library than [`toml`](https://pypi.org/project/toml/). The PR as written will require users to install `toml` to use `pyproject.toml`, but the changes look sufficient to keep it from exploding on a user who doesn't have it.

Reply:
Good looking out - looks like `mypy` switch to requiring `tomli` rather than `toml` since this PR was submitted (https://github.com/python/mypy/commit/aede3aefd75332f3911a43bb4a6a98baacce785a).  We should definitely change that dependency.

Reply:
added in 70d85fc

Reply:
actually, looks like that's not released yet. 0.910 still requires `toml`.  I'll add some checks around that.

Reply:
OK, it'll default to `toml` now, fall back to `tomli` for upcoming releases, and error out otherwise now, so the bases should be covered.